src/FixedDataTableCellGroup.js

```diff
@@ -76,25 +87,70 @@ var FixedDataTableCellGroupImpl = React.createClass({
     }, false);
 
     var currentPosition = 0;
-    for (var i = 0, j = columns.length; i < j; i++) {
+    var count = 0;
+
+    var newColumnsToRender = new Array(props.maxVisibleColumns);
+    var positions = new Array(props.maxVisibleColumns);
+
+    for (var i = 0; i < columns.length; i++) {
       var columnProps = columns[i].props;
       var recycable = columnProps.allowCellsRecycling && !isColumnReordering;
       if (!recycable || (
```

Comment:
Definitely going to need a second review for everything below this line, for all the new logic. Please make it so that variables are named descriptively, loops and other non self-evident code is commented, esp. with regard to the operations performed on your arrays of indexes.

Right now, most variables need to be individually studied to be interpreted, and each line of code reread several times. I'm sure the edge cases will be better identified by everyone if we can get the code here to a place where it's more easily readable.